roles/airgap/tasks/main.yml

```
@@ -81,21 +81,22 @@
         mode: "0755"
         state: directory
 
-    - name: Distribute K3s images {{ k3s_arch }}
+    - name: Gather image archives
+      find:
+        paths: "{{ airgap_dir }}"
+        patterns: "*.tar*"
+      register: found_images
+      delegate_to: localhost
+      become: false
+
+    - name: Distribute image archives
```

You don't need to break this out into two steps, this can be accomplished with a slight modification to the original copy, while ensuring non images don't get placed and cause issues.

```suggestion
    - name: Distribute Airgap images {{ k3s_arch }}
      ansible.builtin.copy:
        src: "{{ item }}"
        dest: /var/lib/rancher/k3s/agent/images/{{ item | basename }}
        owner: root
        group: root
        mode: "0755"
      with_fileglob:
        - "{{ airgap_dir }}/*.tar.gz"
        - "{{ airgap_dir }}/*.tar.zst"
        - "{{ airgap_dir }}/*.tar"
```